ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/SubstringTest.java

```diff
@@ -32,38 +34,83 @@ public void setUp() {
   }
 
   @Test
-  public void shouldReturnNullOnNullValue() {
-    assertThat(udf.substring(null, 1), is(nullValue()));
-    assertThat(udf.substring(null, 1, 1), is(nullValue()));
+  public void shouldReturnNullOnStringNullValue() {
+    assertThat(udf.substring((String) null, 1), is(nullValue()));
+    assertThat(udf.substring((String) null, 1, 1), is(nullValue()));
     assertThat(udf.substring("some string", null, 1), is(nullValue()));
     assertThat(udf.substring("some string", 1, null), is(nullValue()));
   }
 
   @Test
-  public void shouldUseOneBasedIndexing() {
+  public void shouldReturnNullOnBytesNullValue() {
+    assertThat(udf.substring((ByteBuffer) null, 1), is(nullValue()));
+    assertThat(udf.substring((ByteBuffer) null, 1, 1), is(nullValue()));
+    assertThat(udf.substring("some string", null, 1), is(nullValue()));
```

Comment:
The first argument should be a ByteBuffer

Reply:
Done